src/main/java/Duke.java

```diff
@@ -1,10 +1,47 @@
+import commands.Command;
+
+import exceptions.InvalidInputException;
+import tasks.TaskList;
+import utils.Parser;
+import utils.Storage;
+import utils.Ui;
+
 public class Duke {
+
+    private Storage storage;
+    private TaskList taskList;
+    private Ui ui;
+
+    public Duke(String filePath) {
+        this.storage = new Storage(filePath);
+        this.ui = new Ui();
+        this.taskList = new TaskList(this.storage.load());
+
+    }
+
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        new Duke("./data/duke.txt").run();
+    }
+
+    /**
+     * Run Duke ChatBot
+     */
+    public void run() {
+        ui.printWelcomeMessage();
+
+        boolean notExitCommand = true;
+
+        while(notExitCommand) {
+            try {
+                String userCommand = ui.readCommand();
+                Command command = Parser.parse(userCommand);
+                notExitCommand = command.execute(this.taskList, this.ui, this.storage);
+                this.storage.save(this.taskList.getFormattedData());
+
+            } catch (InvalidInputException e) {
+                ui.printError(e.getMessage());
+            }          
+        }
+        ui.printExitMessage();
```

Comment:
Should this be encapsulated within `ExitCommand::execute`?

Reply:
Nice catch! Execute handling the message does make more sense!